### PR TITLE
[Codesample][Updatetable][1647106]

### PR DESCRIPTION
**Pull request by KirtiPereira**

### CODE REVIEW

File: msteams-platform/bots/how-to/conversations/send-proactive-messages.md [hunk 1]
```diff
@@ -233,14 +233,14 @@ You must supply the user ID and the tenant ID. If the call succeeds, the API ret
...
+| **Sample Name** | **Description** | **.NET** | **JavaScript** | **Python** |
+|---------------|--------------|--------|-------------|--------|
+| Teams Conversation Basics  | Demonstrates basics of conversations in Teams, including sending one-to-one proactive messages.| [.NET&nbsp;Core](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/57.teams-conversation-bot) | [JavaScript](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/57.teams-conversation-bot) | [Python](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/python/57.teams-conversation-bot) |
+| Start new thread in a channel | Demonstrates creating a new thread in a channel. | [.NET&nbsp;Core](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel) | [JavaScript](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/javascript_nodejs/58.teams-start-new-thread-in-channel) | [Python](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/python/58.teams-start-thread-in-channel) |
```
Comment (surbhigupta12) [inline msteams-platform/bots/how-to/conversations/send-proactive-messages.md:243]: Please change to View in column 3

**Review by surbhigupta12 — CHANGES_REQUESTED** [review]
Minor comment

File: msteams-platform/bots/how-to/conversations/send-proactive-messages.md (hunk 1 above)
Comment (v-pmuthyala) [reply msteams-platform/bots/how-to/conversations/send-proactive-messages.md]: changed to View

File: msteams-platform/includes/messaging-extensions/learn-more.md [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
-  * [Messaging extension with search-based commands](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/50.teams-messaging-extensions-search)
\ No newline at end of file
+  | **Sample name** | **Description** | **.NET** | **Node.js** |
+  |---------------|---------------|-----------|----------|
+  |  Messaging extension with action-based commands | This sample illustrates how to build an Action-based Messaging Extension. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/51.teams-messaging-extensions-action) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/51.teams-messaging-extensions-action) |
+  | Messaging extension with search-based commands | This sample illustrates how to build a Search-based Messaging Extension. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/50.teams-messaging-extensions-search) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/50.teams-messaging-extensions-search) |
```
Comment (Wajeed-msft) [inline msteams-platform/includes/messaging-extensions/learn-more.md:8]: Could you please add Python reference as well?

**Search-** https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/50.teams-messaging-extension-search
**Action-** https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/51.teams-messaging-extensions-action

File: msteams-platform/messaging-extensions/how-to/add-configuration-page.md [hunk 3]
```diff
@@ -9,9 +9,14 @@ ms.author: anclear
...
 [!include[v4-to-v3-SDK-pointer](~/includes/v4-to-v3-pointer-me.md)]
 
-## Bot Framework Samples
+## Code sample
+
+The Teams Search Auth Config for sample projects on GitHub, demonstrate how to create messaging extensions that include a configuration page and Bot Service authentication. The samples also demonstrate how to create message extensions that accept search requests and return the results after the user has signed in.
+
+| **Sample name** | **Description** | **.NET** | **Node.js** |
+|-----------------|-----------------|-------------|--------------|
+| Bot builder | To create messaging extensions. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/52.teams-messaging-extensions-search-auth-config) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/52.teams-messaging-extensions-search-auth-config) |
```
Comment (Wajeed-msft) [inline msteams-platform/messaging-extensions/how-to/add-configuration-page.md:18]: Include python link as well: https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/50.teams-messaging-extension-search

File: msteams-platform/messaging-extensions/what-are-messaging-extensions.md [hunk 4]
```diff
@@ -54,16 +54,14 @@ They can be triggered from the compose message area or the command box. Unlike a
...
-  * [Messaging extension with search-based commands](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/50.teams-messaging-extensions-search)
-* **JavaScript**
-  * [Messaging extension with action-based commands](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/51.teams-messaging-extensions-action)
-  * [Messaging extension with search-based commands](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/50.teams-messaging-extensions-search)
+| **Sample name** | **Description** | **.NET** | **Node.js** |
+|------------|-------------|----------------|------------|
+| Messaging extension with action-based commands | This sample illustrates how to build an action-based messaging extension. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/51.teams-messaging-extensions-action) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/51.teams-messaging-extensions-action) |
```
Comment (Wajeed-msft) [inline msteams-platform/messaging-extensions/what-are-messaging-extensions.md:63]: Could you please add Python reference as well?

**Search-** https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/50.teams-messaging-extension-search
**Action-** https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/51.teams-messaging-extensions-action

File: msteams-platform/samples/virtual-assistant.md [hunk 5]
```diff
@@ -541,12 +541,16 @@ Book-a-room bot is designed for private chats (personal/1:1 scope) only. Since w
...
+You can also leverage existing skills from [Bot Framework Solutions repository](https://github.com/microsoft/botframework-solutions/tree/master/skills/csharp) or create a new skill altogether from scratch. Tutorials for the later can be found [here](https://microsoft.github.io/botframework-solutions/overview/skills/). Please refer to [documentation](https://docs.microsoft.com/azure/bot-service/skills-conceptual?view=azure-bot-service-4.0&preserve-view=true) for Virtual Assistant and skills architecture.
+
+## Code sample
+
+| **Sample name** | **Description** | **C#** | **.NET** |
+|----------|-----------------|----------|------------------|
+| Updated visual studio template | Customized template to support teams capabilities. | [View](https://github.com/OfficeDev/microsoft-teams-apps-bookaroom/tree/nebhagat/microsoft-teams-apps-bookaroom-skill) |
+| Book-a-room bot skill code | Lets you quickly find and book a meeting room on the go. |  | [View](https://github.com/nebhagat/msteams-virtual-assistant-dotnet) |
```
Comment (Wajeed-msft) [inline msteams-platform/samples/virtual-assistant.md:551]: We should not include reference to personal repositories in documentation. Could you please get in touch with Nehal Bhagat and see if this sample is available in OfficeDev repo? Also, C# .NET should be single column.

File: msteams-platform/tabs/how-to/authentication/auth-flow-tab.md [hunk 6]
```diff
@@ -38,12 +38,13 @@ Similar to other application auth flows in Teams, the start page must be on a do
...
+## Code sample
 
-For sample code showing the tab authentication process, see:
+Sample code showing the tab authentication process.
 
-* [Teams tab authentication sample (Node)](https://github.com/OfficeDev/microsoft-teams-sample-complete-node)
-* [Teams tab authentication sample (C#)](https://github.com/OfficeDev/microsoft-teams-sample-complete-csharp)
+| **Sample name** | **Description** | **C#** | **Node.js** |
+|-----------------|-----------------|-------------|------------|
+| Teams tab authentication | Authentication process for tabs using AAD. | [View](https://github.com/OfficeDev/microsoft-teams-sample-complete-csharp) | [View](https://github.com/OfficeDev/microsoft-teams-sample-complete-node) |
```
Comment (Wajeed-msft) [inline msteams-platform/tabs/how-to/authentication/auth-flow-tab.md:47]: Could you please update the links? 

C#  - https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-complete-sample/csharp
Node.js - https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-complete-sample/nodejs

File: msteams-platform/tabs/how-to/authentication/auth-tab-aad.md [hunk 7]
```diff
@@ -148,8 +148,10 @@ Your app can set its own session cookie so that the user need not sign in again
...
-## Samples
+## Code sample
 
-For sample code showing the tab authentication process using Azure AD see:
+Sample code showing the tab authentication process using Azure AD:
 
-* [Microsoft Teams tab authentication sample (Node)](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
+| **Sample name** | **description** | **Node.js** |
+|-----------------|-----------------|-------------|
+| Microsoft Teams tab authentication | Tab authentication process using Azure AD. | [View](https://github.com/OfficeDev/microsoft-teams-sample-auth-node) |
```
Comment (Wajeed-msft) [inline msteams-platform/tabs/how-to/authentication/auth-tab-aad.md:157]: Could you please update the link to following? 

https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-auth/nodejs

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: msteams-platform/samples/virtual-assistant.md (hunk 5 above)
Comment (nebhagat) [reply msteams-platform/samples/virtual-assistant.md:551]: Unfortunately we don't have this repo available in OfficeDev org. @aosolis do you think we should move this repo? 
Till then please refer to my personal repo.

Comment (aosolis) [reply msteams-platform/samples/virtual-assistant.md:551]: @nebhagat Can you work with Wajeed to post this to the samples repo that he maintains?

Comment (nebhagat) [reply msteams-platform/samples/virtual-assistant.md:551]: sure, will reach out to Wajeed

File: msteams-platform/bots/how-to/authentication/auth-flow-bot.md [hunk 8]
```diff
@@ -43,11 +43,13 @@ for an example that demonstrates authentication flow for bots using Node.js and
...
-## Samples
+## Code sample
 
-For sample code showing the bot authentication process see:
+Sample code showing the bot authentication process:
 
-[Microsoft Teams bot authentication sample (Node.js)](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
+| **Sample name** | **Description** | **Node.js** |
+|-----------------|----------------|--------------|
+| Microsoft Teams bot authentication sample | This sample demonstrates authentication in Microsoft Teams apps. | [View](https://github.com/OfficeDev/microsoft-teams-sample-auth-node) |
```
Comment (Wajeed-msft) [inline msteams-platform/bots/how-to/authentication/auth-flow-bot.md:52]: Could you please also link all the three samples listed here?

https://github.com/microsoft/BotBuilder-Samples#teams-samples
image

File: msteams-platform/concepts/authentication/authentication.md [hunk 9]
```diff
@@ -31,7 +31,14 @@ For more information on using the OAuthPrompt, see:
...
 * [Bot authentication flow overview](~/bots/how-to/authentication/auth-flow-bot.md) describes how authentication works within a bot in the app in Teams. This shows a non-web-based authentication flow used for bots on Teams web, desktop app, and mobile apps.
 * [Bot authentication](~/bots/how-to/authentication/add-authentication.md) describes how to add OAuth authentication to the Teams bot.
-* [.Net or C#](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/46.teams-auth) or [JavaScript or Node.js](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/46.teams-auth) provides Bot authentication v3 SDK sample.
+
+## Code sample
+
+provides Bot authentication v3 SDK sample.
+
+| **Sample name** | **Description** | **.NET** | **Node.js** |
+|---------------|------------|------------|-------------|
```
Comment (Wajeed-msft) [inline msteams-platform/concepts/authentication/authentication.md:40]: Link Python sample as well: https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/46.teams-auth

File: msteams-platform/tabs/how-to/authentication/auth-tab-aad.md [hunk 10]
```diff
@@ -148,8 +148,10 @@ Your app can set its own session cookie so that the user need not sign in again
...
-## Samples
+## Code sample
 
-For sample code showing the tab authentication process using Azure AD see:
+Sample code showing the tab authentication process using Azure AD:
 
-* [Microsoft Teams tab authentication sample (Node)](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
+| **Sample name** | **description** | **Node.js** |
+|-----------------|-----------------|-------------|
+| Microsoft Teams tab authentication | Tab authentication process using Azure AD. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-auth/nodejs) |
```
Comment (Wajeed-msft) [inline msteams-platform/tabs/how-to/authentication/auth-tab-aad.md:157]: Please link .NET  sample: https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/tab-channel-group-config-page-auth/csharp

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by surbhigupta12 — APPROVED** [review]
Approved

**Pull request merged**